esiclient/v1/switch.py

```diff
@@ -55,9 +63,11 @@ def take_action(self, parsed_args):
                        if next(iter(np.fixed_ips), None).get(
                                'subnet_id', None) == subnet_id)
                 for np in nps:
+                    search_np_id = subnp_np_map.get(np.id, np.id)
```

Comment:
I don't understand this part. Why do we get the neutron port id from subnp_np_map?

Reply:
I've added a comment to try and explain (I should be adding more comments). But basically, this is needed for trunk subports, as they don't directly have the information mapping them to a particular switchport.

Reply:
Oh, got it!